sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/ViewBuilder.java

```diff
@@ -52,19 +56,102 @@ public ViewBuilder setAggregatorFactory(AggregatorFactory aggregatorFactory) {
   }
 
   /**
-   * sets {@link LabelsProcessorFactory}.
+   * Specify the attributes processor for this view.
    *
-   * @param labelsProcessorFactory labels processor factory.
+   * <p>Note: This resets all attribute filters, baggage appending and other processing.
+   *
+   * @param processor The pre-processor for measurement attributes.
+   * @return this Builder.
+   */
+  public ViewBuilder setAttributesProcessor(AttributesProcessor processor) {
```

Comment:
My preference would be to keep the surface area of configuring the attributes processor smaller, with just two methods:
```
public ViewBuilder setAttributesProcessor(AttributesProcessor processor);

public ViewBuilder addAttributeProcessor(AttributesProcessor processor) 
```

Reply:
AttributesProcessor isn't specified.   I'd agree that this is simpler, but my goal was NOT to expose attribute processor publicly for two reasons:

1. I expect refactoring in the aggregator API (and a specification at some point).   It's likely this API would break at that point.
2.  We've been avoiding placing anything in a stable location that is unspecified (or where we're not just deviating, but inventing new functionality that diverges from the spec).